Storage/src/Lifecycle.php

```diff
@@ -0,0 +1,236 @@
+     *     The condition(s) where the rule will apply.
+     *
+     *     @type int $age Age of an object (in days). This condition is
+     *           satisfied when an object reaches the specified age.
+     *     @type string $createdBefore A date in RFC 3339 format with only the
+     *           date part (for instance, "2013-01-15"). This condition is
+     *           satisfied when an object is created before midnight of the
+     *           specified date in UTC.
+     *     @type bool $isLive Relevant only for versioned objects. If the value
+     *           is `true`, this condition matches live objects; if the value is
+     *           `false`, it matches archived objects.
+     *     @type string[] $matchesStorageClass Objects having any of the storage
+     *           classes specified by this condition will be matched. Values
+     *           include `"MULTI_REGIONAL"`, `"REGIONAL"`, `"NEARLINE"`,
+     *           `"COLDLINE"`, `"STANDARD"`, and `"DURABLE_REDUCED_AVAILABILITY"`.
+     *     @type int $numNewerVersions Relevant only for versioned objects. If
+     *           the value is N, this condition is satisfied when there are at
+     *           least N versions (including the live version) newer than this
+     *           version of the object.
+     * }
+     * @return Lifecycle
+     */
+    public function addSetStorageClassRule($storageClass, array $condition)
+    {
+        $this->lifecycle['rule'][] = [
+            'action' => [
+                'type' => 'SetStorageClass',
+                'storageClass' => $storageClass
+            ],
+            'condition' => $condition
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Clears out rules based on the provided condition.
+     *
+     * Example:
+     * ```
+     * // Remove all rules.
+     * $lifecycle->clearRules();
+     * ```
+     *
+     * ```
+     * // Remove all "Delete" based rules.
+     * $lifecycle->clearRules('Delete');
+     * ```
+     *
+     * @param string|callable $condition [optional] If a string is provided, it
+     *        must be the name of the type of rule to remove (`SetStorageClass`
+     *        or `Delete`). All rules of this type will then be cleared. When
+     *        providing a callable you may define a custom route for how you
+     *        would like to remove rules. The provided callable will be run
+     *        through
+     *        [array_filter](http://php.net/manual/en/function.array-filter.php).
+     *        The callable's argument will be a single lifecycle rule as an
+     *        associative array. When returning true from the callable the rule
+     *        will be preserved, and if false it will be removed.
+     *        **Defaults to** `null`, clearing all assigned rules.
+     * @return Lifecycle
+     * @throws \InvalidArgumentException If a type other than a string or
+     *         callabe is provided.
+     */
+    public function clearRules($condition = null)
```

Comment:
parameter name might be confusing with an OLM condition. What about `$action`?

Reply:
SGTM